### PR TITLE
task: Query default branch via api

### PR DESCRIPTION
**Pull request by Gundersanne**
This is a requirement to make pull requests created by scripts work with
different default branches.

----

`osbuild/image-builder-frontend` recently switched to `main` as a default branch and the `npm-update` workflow stopped working because it would try to create a PR against a branch which doesn't exist. With this patch it'll query the default branch from the api.

### CODE REVIEW

**Comment by Gundersanne** [comment]
forced an npm-update run: https://github.com/osbuild/image-builder-frontend/pull/61

File: task/github.py [hunk 1]
```diff
@@ -169,14 +169,16 @@ def url(self):
...
+                self._base = "{0}/repos/{1}".format(netloc, self.repo)
 
             self._url = urllib.parse.urlparse(self._base)
 
         return self._url
 
     def qualify(self, resource):
-        return urllib.parse.urljoin(self.url.path, resource)
+        if resource is None:
+            return self.url.path
```
Comment (martinpitt) [inline task/github.py:180]: That looks unrelated -- what's that for? Should it be a separate commit?

**Review by martinpitt — COMMENTED** [review]
Thanks, this is nice! Just one question.

File: task/github.py (hunk 1 above)
Comment (Gundersanne) [reply task/github.py:180]: ah yea, this should be a separate commit. Doing that :)

Comment (Gundersanne) [reply task/github.py:180]: Wait on second thought it shouldn't be, but it is kinda confusing. I can add a bit more explanation when squashing.

So `/repos/:owner/:repo` is also an endpoint, this has the `default_branch` value in it. And `qualify()` always appends an `/`, which breaks that particular endpoint.

This way `api.get()` will query the `/repos/:owner/:repo` endpoint.

Comment (martinpitt) [reply task/github.py:180]: Thanks. I can't say that I fully understand this, but fine to land then with some extra explanation.

**Review by martinpitt — APPROVED** [review]

File: task/github.py (hunk 1 above)
Comment (Gundersanne) [reply task/github.py:180]: So an api call to `/repos/cockpit-project/cockpit` returns an object; an api call to `/repos/cockpit-project/cockpit/` returns None

Comment (martinpitt) [reply task/github.py:180]: Yes, I get that part, that's line 172. I meant the default fallback of "resource" here.
